### PR TITLE
[Fix] Tensorflow MASTER implementation

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- refactored tensorflow transformer implementation
- fix MASTER Decoder

Any feedback is welcome :hugs: 
@charlesmindee @frgfm feel free to take a first look some improvements are very welcome :D

Todo:

- [x] toy benchmark (train: 500k MJSynth  / val: FUNSD/CORD)
- [x] fix decoding stage while inference (Note: training works well only decoding troublesome)
- [ ] replace numpy cast in positional encoding
Issue:
#802

### CODE REVIEW

**Comment by felixdittrich92** [comment]
@frgfm If you should get to it before me, please take a look at the decoding and code improvements in advance would also be quite good 👍

**Comment by felixdittrich92** [comment]
@frgfm this PR would be very happy to get your help 🤣

File: doctr/models/recognition/transformer/tensorflow.py [hunk 1]
```diff
@@ -3,263 +3,177 @@
...
+            tf.expand_dims(tf.experimental.numpy.arange(start=0, stop=max_len), axis=1), dtype=tf.float32
+        )
+        div_term = tf.math.exp(
+            tf.cast(
+                tf.experimental.numpy.arange(start=0, stop=d_model, step=2), dtype=tf.float32
+            ) * -(math.log(10000.0) / d_model)
+        )
+        pe = pe.numpy()
+        pe[:, 0::2] = tf.math.sin(position * div_term)
+        pe[:, 1::2] = tf.math.cos(position * div_term)
```
Comment (felixdittrich92) [inline doctr/models/recognition/transformer/tensorflow.py:38]: @frgfm Any idea how to do this in TF and remove the numpy cast? :sweat_smile:

Comment (charlesmindee) [reply doctr/models/recognition/transformer/tensorflow.py:38]: Hi @felixdittrich92, I know [this](https://www.tensorflow.org/api_docs/python/tf/numpy_function) but I think it is still a bit experimental

**Review by charlesmindee — APPROVED** [review]
Thanks, LGTM!

**Review by charlesmindee — APPROVED** [review]
Thanks! LGTM

**Pull request merged**